File: Task-management/task-management-ui/src/services/taskService.js
import API from "./api";

export const getTasks = () => API.get("/tasks");

export const getTask = (id) => API.get(`/tasks/${id}`);

export const createTask = (task) => API.post("/tasks", task);

export const updateTask = (id, task) => API.put(`/tasks/${id}`, task);

export const deleteTask = (id) => API.delete(`/tasks/${id}`);

export const searchTasks = (keyword) =>
    API.get(`/tasks/search?keyword=${keyword}`);

export const getTasksByProject = (projectId) =>
    API.get(`/tasks/project/${projectId}`);

File: Task-management/task-management/src/main/java/com/taskmanagement/service/TaskService.java
package com.taskmanagement.service;

import com.taskmanagement.dto.TaskRequest;
import com.taskmanagement.dto.TaskStatsResponse;
import com.taskmanagement.entity.Task;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TaskService {

    Task createTask(TaskRequest request);

    List<Task> getAllTasks();

    Task getTask(Long id);

    Task updateTask(Long id, TaskRequest request);

    void deleteTask(Long id);

    TaskStatsResponse getTaskStats();

    List<Task> searchTasks(String keyword);

    List<Task> getTasksByStatus(String status);

    List<Task> getTasksByPriority(String priority);

    // ADD THIS METHOD
    List<Task> getHighPriorityTasks();

    List<Task> getTasksByUser(Long userId);

    List<Task> getTasksByProject(Long projectId);

    Page<Task> getTasks(int page, int size, String sortBy);

    List<Task> sortTasks(String sortBy);
}